src/store_client.cc

```diff
@@ -622,6 +639,9 @@ store_client::readHeader(char const *buf, ssize_t len)
     if (len < 0)
         return fail();
 
+    if (ale)
+        ale->replyBytesFromDisk += len;
```

Comment:
I do not think so. Given the two options, I think it is best to report how many bytes were read in total, regardless of the final outcome. Otherwise, effectively, we would have to keep answering (in code and documentation) a rather complex question: "Which read bytes were actually _used_ to form the final response?" and reset counters whenever we think the bytes read from the store were left unused for the final response. For an extreme hypothetical example, imagine an error page that actually quotes/includes some failed revalidation response headers (read from Store before the error response was generated by Squid).

We already document that we add up primary and revalidation bytes. I added more text to cover the answer to your question in commit 16c6af2.